### ISSUE TITLE
Error and DC on buyauto equipment

### ISSUE THREAD

**Issue by Henrybk** [OpenKore/openkore#806]
When buyauto is set to buy an weapon or armor from a npc the packet which must be sent to server must contain the desired item's ID ((as in 601 for Fly Wing), kore tries to get the item ID based on it's own inventory and on %items_lut, this can be problematic as it does not check the npc sell list and can send wrong packets to the server.

I just experinced this with one of my bots trying to buy a Damascus [1] from the morroc weapon dealer.

buyauto block:
```
buyAuto Damascus {
   npc morocc_in 141 67
   minAmount 0
   maxAmount 1
   zeny  > 55000
   inInventory Damascus < 1
}
```

Npc Sell List:
```
---------------- Store List (Unknown) ----------------
#  Name                    Type                  Price
0  Arrow                   Arrow                    1z
1  Silver Arrow            Arrow                    3z
2  Bow                     Pet Armor             1000z
3  Rod                     Weapon                  50z
4  Knife                   Weapon                  50z
5  Cutter                  Weapon                1250z
6  Main Gauche             Weapon                2400z
7  Dirk                    Weapon                8500z
8  Dagger                  Weapon               14000z
9  Stiletto                Weapon               19500z
10 Gladius                 Weapon               43000z
11 Damascus                Weapon               49000z
12 Jur                     2-Hand Weapon        19500z
13 Katar                   2-Hand Weapon        41000z
14 Jamadhar                2-Hand Weapon        37200z
------------------------------------------------------
```

Dumper of the $args used to send the packet:
```
$VAR1 = {
          'binID' => undef,
          'distance' => undef,
          'sentBuy' => 1,
          'itemID' => '1226',
          'npc' => {
                     'ok' => 1,
                     'map' => 'morocc_in',
                     'pos' => {
                                'y' => 67,
                                'x' => 141
                              }
                   },
          'index' => 3,
          'lastIndex' => 3,
          'index_failed' => {}
        };
```

The Id of `Damascus [1]` which the npc does sell is `1222` but openkore was trying to buy a `Damascus [2]` which has Id `1226` because it was found first in %items_lut keys search.

While looking at aech key in %items_lut is very slow it also can cause bugs like these, I suggest we check for the item in the actual Npc sell list of event allow users to set buyAuto blocks with item's IDs istead of names.

**Comment by rhnbotfelt11** [comment]
pusing mau ngebot? Beli Zeny Mahal? Jasa joki juga mahal? mending ngebot sendiri aman bisa gb 100id ++ sampe lvl 99 bisa ngumpulin zeny cepat bahakan bisa jualan zeny.

Jual BOT Ragnarok Hanya 300Ribu berminat hubungi:081380644602 Mumpung harga promo

https://www.youtube.com/watch?v=8xLIUQH3mKM

**Comment by idrhnbot1** [comment]
Ayoo gan promo tinggal untuk 1 orang lagi sudah ada yg order 

Bukti pembelian promo kemarin:
https://image.ibb.co/b3buzk/IMAG0170.jpg

BOT Ragnarok No Limit No HWID bisa ngebot 100 ID++ dalam 1PC bisa di pakai PC laptop warnet tanpa batas.

Dari pada beli Zeny mending jualan Zeny,Dari pada Bayar jasa Joki Mahal Mending Joki pakai BOT bisa bikin lvl 99 100ID++ tanpa bayar joki

berminat hubungi:
081380644602

www.rhanfelta.blogspot.com

**Comment by alisonrag** [comment]
Fixed in #809
